services/inference/tsfminference/chronos_service_handler.py

```diff
@@ -0,0 +1,134 @@
+
+    """
+
+    def __init__(
+        self,
+        model_id: str,
+        model_path: Union[str, Path],
+        handler_config: TSFMConfig,
+    ):
+        # !!! Double check which init
+        super().__init__(
+            model_id=model_id, model_path=model_path, handler_config=handler_config
+        )
+
+    def _prepare(
+        self,
+        data: pd.DataFrame,
+        future_data: Optional[pd.DataFrame] = None,
+        schema: Optional[ForecastingMetadataInput] = None,
+        parameters: Optional[ForecastingParameters] = None,
+        **kwargs,
+    ) -> "ChronosForecastingHandler":
+        """Implementation of _prepare for Chronos family of models. This method loads the model using chronos apis.
+
+        Args:
+            data (pd.DataFrame): A pandas dataframe containing historical data.
+            future_data (Optional[pd.DataFrame], optional): A pandas dataframe containing future data. Defaults to None.
+            schema (Optional[ForecastingMetadataInput], optional): Schema information from the original inference
+                request. Includes information about columns and their role. Defaults to None.
+            parameters (Optional[ForecastingParameters], optional): Parameters from the original inference
+                request. Defaults to None.
+
+        Returns:
+            ChronosForecastingHandler: The updated service handler object.
+        """
+
+        model = ChronosPipeline.from_pretrained(
+            "amazon/chronos-t5-tiny",
+        )
+        self.model = model
+        self.config = model.model.model.config
+        self.chronos_config = self.config.chronos_config
+
+        return self
+
+    def _run(
+        self,
+        data: pd.DataFrame,
+        future_data: Optional[pd.DataFrame] = None,
+        schema: Optional[ForecastingMetadataInput] = None,
+        parameters: Optional[ForecastingParameters] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        """Inference request for Chronos family of models.
+
+        Args:
+            data (pd.DataFrame): Input historical time series data.
+            future_data (Optional[pd.DataFrame], optional): Input future time series data, useful for
+                passing future exogenous if known. Defaults to None.
+            schema (Optional[ForecastingMetadataInput], optional): Schema information from the original inference
+                request. Includes information about columns and their role. Defaults to None.
+            parameters (Optional[ForecastingParameters], optional): Parameters from the original inference
+                request. Defaults to None.
+
+        Returns:
+            pd.DataFrame: The forecasts produced by the model.
+        """
+
+        target_columns = schema.target_columns or data.columns.tolist()
+        prediction_length = (
+            parameters.prediction_length or self.chronos_config["prediction_length"]
+        )
+
+        num_samples = self.chronos_config["num_samples"]
```

Comment:
Can these be part of parameters?

Reply:
Current version of service APIs do no allow model specific run time parameters. I have created #204  to track.